contracts/exchange/contracts/src/MixinSignatureValidator.sol

```diff
@@ -305,6 +350,53 @@ contract MixinSignatureValidator is
         ));
     }
 
+    /// @dev Verifies signature using logic defined by an EIP1271 Wallet contract.
+    /// @param hash Any 32 byte hash.
+    /// @param walletAddress Address that should have signed the given hash
+    ///                      and defines its own signature verification method.
+    /// @param signature Proof that the hash has been signed by signer.
+    /// @return True if the signature is validated by the Walidator.
+    function _validateHashWithEIP1271Wallet(
```

Comment:
This signature type doesn't feel necessary to me. The hash can always be derived from the data, so I guess this is just a cheaper alternative?

Reply:
Yeah. I was mainly aiming for parity with existing signature types. I would be in favor of removing this signature type, but my concern is what if we want to use EIP-1271 with `ZeroExTransaction`s, which currently only pass a hash?